Hunting Queries/AuditLogs/AccountAddedtoPrivilegedPIMGroup

```diff
@@ -0,0 +1,29 @@
+name: AccountAddedtoPrivilegedPIMGroup
+description: |
+  'Identifies accounts that have been added to a PIM managed privileged group'
+requiredDataConnectors:
+  - connectorId: Azure Active Directory
+    dataTypes:
+      - AuditLogs
+tactics:
+  - Persistence
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1098
+  - T1548
+query: |
+AuditLogs
+| where TimeGenerated > ago(1d)
+| where ActivityDisplayName =~ "Add eligible member to role in PIM requested (timebound)"
+| where AADOperationType =~ "CreateRequestEligibleRole"
+| where TargetResources contains "-PRIV"
+or TargetResources contains "Administrator"
+or TargetResources contains "Security"
+| extend BuiltinGroup= tostring(parse_json(TargetResources[0].displayName))
+| extend CustomGroup= tostring(parse_json(TargetResources[3].displayName))
+| extend TargetAccount_ = tostring(parse_json(TargetResources[2].displayName))
+| extend Initiatedby = tostring(parse_json(Identity))
```

Comment:
Parsing needs to be changed to handle dynamic arrays and avoids creating duplicate columns. 
please review below version and make changes. 

```
| mv-expand TargetResources
| extend  DisplayNames = tostring(parse_json(TargetResources).displayName), Type = tostring(parse_json(TargetResources).type)
| extend InitiatedbyDisplayName = tostring(parse_json(parse_json(InitiatedBy).user).displayName) , UserPrincipalName = tostring(parse_json(parse_json(InitiatedBy).user).userPrincipalName)
```

Reply:
Chose to not do this as it make results less clear on what role and group a user was added to.